static-analysis/StaticAnalysisChapter.md

```diff
@@ -880,15 +882,15 @@ check_locals(f::Callable) = all([check_locals(e) for e in code_typed(f)])
 check_locals(e::Expr) = isempty(unused_locals(e))
 ~~~
 
-# Conclusion
-We’ve done two static analyses of Julia code -- one based on types and one based on variable usages.
+## Conclusion
+We’ve done two static analyses of Julia code --- one based on types and one based on variable usages.
 
 Statically-typed languages already do the kind of work our type-based analysis did; additional type-based static analysis is mostly useful in dynamically typed languages. There have been (mostly research) projects to build static type inference systems for languages including Python, Ruby, and Lisp. These systems are usually built around optional type annotations; you can have static types when you want them, and fall back to dynamic typing when you don’t. This is especially helpful for integrating some static typing into existing code bases.
 
-Non-typed-based checks, like our variable-usage one, are applicable to both dynamically- and statically-typed languages. However, many statically-typed languages, like C++ and Java, require you to declare variables and already give basic warnings like the ones we created. There are still custom checks that can be written. For example, checks that are specific to your project’s style guide or extra safety precautions based on security policies.
+Non-typed-based checks, like our variable-usage one, are applicable to both dynamically and statically typed languages. However, many statically typed languages, like C++ and Java, require you to declare variables, and already give basic warnings like the ones we created. There are still custom checks that can be written; for example, checks that are specific to your project’s style guide or extra safety precautions based on security policies.
 
-While Julia does have great tools for enabling static analysis, it’s not alone. Lisp, of course, is famous for having the code be a data structure of nested lists, so it tends to be easy to get at the AST. Java also exposes it’s AST, although the AST is much more complicated than Lisp’s. Some languages or language tool-chains are not designed to allow mere users to poke around at internal representations. For open-source tool chains (especially well-commented ones), one option is to add the hooks you want to pull out the AST.
+While Julia does have great tools for enabling static analysis, it’s not alone. Lisp, of course, is famous for having the code be a data structure of nested lists, so it tends to be easy to get at the AST. Java also exposes its AST, although the AST is much more complicated than Lisp’s. Some languages or language tool-chains are not designed to allow mere users to poke around at internal representations. For open-source tool chains (especially well-commented ones), one option is to add the hooks you want to pull out to the AST.
```

Comment:
No. We're not adding anything to the AST. The hooks get added to the language/compiler/etc; the hooks pull out the AST for us.